### DISCUSSION TITLE
Copy a file to (a) node(s)

### DISCUSSION THREAD

**Discussion by oneindelijk** [choria-io/general#34] [Q&A] [answered]
Hi, 

I've been looking in the documentation and the internet, but I couldn't figure out which approach to use in order to copy a file from the client onto one or several servers.
I'm suspecting it should be possible with the puppet file module, but I don't know how to implement,
some guidance would be helpful

Am I trying to use choria to much as I would use Ansible?

**Comment by ripienaar** [comment, marked as answer]
At present not much to do about that, Choria isnt made for file transfer - you could though initiate some kind of download action that fetch from a bucket, thats more Chorias vibe.

We are exploring some options for object store in the choria broker but its early days.

**Comment by smortex** [comment]
I wrote a "toy" transfer agent/application some times ago, far from ideal because you also have a copy of each file content in the logs :upside_down_face:, but for small files it helped me at some point: https://github.com/smortex/transfer-agent

More recently I needed to transfer huge binary files from / to arbitrary machines, and I used [magic wormhole](https://github.com/magic-wormhole/magic-wormhole) for this. We created a module with bolt tasks, and we use choria to orchestrate these bolt tasks in site plans:
https://forge.puppetlabs.com/modules/opuscodium/wormhole
